samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -240,16 +261,26 @@ CountDownLatch getShutdownLatch() {
   }
 
   @VisibleForTesting
-  StreamProcessor createStreamProcessor(Config config, ApplicationDescriptorImpl<? extends ApplicationDescriptor> appDesc,
+  MetadataStore createCoordinatorStreamStore(Config jobConfig) {
+    MetadataStore coordinatorStreamStore =
+        coordinatorMetadataStoreFactory.getMetadataStore("NoOp", jobConfig, new MetricsRegistryMap());
+    return coordinatorStreamStore;
+  }
+
+  @VisibleForTesting
+  Pair<StreamProcessor, MetadataStore> createStreamProcessor(Config config, ApplicationDescriptorImpl<? extends ApplicationDescriptor> appDesc,
```

Comment:
Just curious. 

1. Why are we passing in the metadata store as a argument to this API and returning it back to the caller of the API again? Would prefer to remove MetadataStore from the result of this API if it makes sense.
 2. Also, please note that LocalApplicationRunner at its current state can only manage a life-cycle of one stream processor. For more details, please look at how the exit of stream processors are handled in LocalApplicationRunner.

Reply:
1. Wanted to pair the store with the processor, but I can do the pairing outside of the result. Will change.
2. Where exacly should I look at? From the looks of the overall code, all the stream processors are kept in a `Set` and iterated over during lifecycle related calls like `kill()`. I don't see where the logic of "only one stream processor lifecycle is managed"

Reply:
>> From the looks of the overall code

Take a look at the life-cycle methods in `LocalApplicationRunner`. A `LocalApplicationRunner` instance can only manage one instance of `StreamApplication`. Launching multiple stream application through a single `LocalApplicationRunner` will lead to internal-state corruption in `LAR`.

 Currently samza only supports single stage jobs. Hence the `Set` you're referring to `LocalApplicationRunner` will be always of size `1`.  

Above point was the motivation behind finding out the rationale behind `pairing` store with a processor.